### ISSUE TITLE
Terminal into Pod is not possible if authentication was disabled

### ISSUE THREAD

**Issue by thewilli** [argoproj/argo-cd#20037]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When authentication to Argo is disabled, you cannot open a Terminal on a Pod.

We use a dedicated, external authentication mechanism and therefore don't require an additional authentication mechanism. Additionally, if you want to setup a local dev system on a [kind cluster](https://kind.sigs.k8s.io/), authentication is not of much use.

**To Reproduce**

- Ensure you can open a Terminal on a Pod / Container
- Set key `server.disable.auth` in the `argocd-cmd-params-cm` ConfigMap to `"true"`
- Restart the `argocd-server` Pod(s)

**Expected behavior**

* A Terminal can still be opened

**Actual behavior**

* The Terminal won't open and fails with an error

**Screenshots**

![CleanShot 2024-09-20 at 16 50 12@2x](https://github.com/user-attachments/assets/1349b57d-8179-4823-a352-94e06acec4a7)

**Version**

```shell
$ argocd version
argocd: v2.12.3+6b9cd82
  BuildDate: 2024-09-10T15:34:33Z
  GitCommit: 6b9cd828c6e9807398869ad5ac44efd2c28422d6
  GitTreeState: clean
  GoVersion: go1.23.1
  Compiler: gc
  Platform: darwin/arm64
argocd-server: v2.12.3+6b9cd82
  BuildDate: 2024-08-27T11:57:48Z
  GitCommit: 6b9cd828c6e9807398869ad5ac44efd2c28422d6
  GitTreeState: clean
  GoVersion: go1.22.4
  Compiler: gc
  Platform: linux/arm64
  Kustomize Version: v5.4.2 2024-05-22T15:19:38Z
  Helm Version: v3.15.2+g1a500d5
  Kubectl Version: v0.29.6
  Jsonnet Version: v0.20.0
```

**Logs**

```
time="2024-09-20T14:55:01Z" level=info msg="terminal session starting" appNamespace=argocd application=test3 container=nginx nam
espace=test3 podName=test3-58c8cb577f-9r6rs project=applications userName=
```

**Comment by zkck** [comment]
Hello, I'd like to take up this issue! :)

**Comment by reggie-k** [comment]
This issue has been fixed for the upcoming release 2.13.0.
